chainer/training/triggers/minmax_value_trigger.py

```diff
@@ -0,0 +1,102 @@
+from chainer import reporter
+from chainer.training.trigger import IntervalTrigger
+
+
+class BestValueTrigger(object):
+
+    """Trigger invoked when specific value becomes best.
+
+    Args:
+        key (str): Key of value.
+        compare (function): Compare function which takes current best value and
+            new value and returns whether new value is better than current
+            best.
+        trigger: Trigger that decides the comparison interval between current
+            best value and new value. This must be a tuple in the form of
+            ``<int>, 'epoch'`` or ``<int>, 'iteration'`` which is passed to
+            :class`IntervalTrigger`.
+
+    """
+    def __init__(self, key, compare, trigger=(1, 'epoch')):
+        self._key = key
+        self._best_value = None
+        self._interval_trigger = IntervalTrigger(*trigger)
+        self._init_summary()
+        self._compare = compare
+
+    def __call__(self, trainer):
+        """Decides whether the extension should be called on this iteration.
+
+        Args:
+            trainer (Trainer): Trainer object that this trigger is associated
```

Comment:
Also I fixed the comment "The updater associated with this trainer is" to "The ``observation`` of this trainer is" to match actual implementation.